fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -458,25 +458,38 @@ void FixedLagSmoother::transactionCallback(
       {
         started_ = true;
         ignited_ = true;
-        start_time = min_time;
-        setStartTime(min_time);
+        start_time = position->minStamp();
+        setStartTime(start_time);
+
+        // And purge out old transactions
+        //  - Either before or exactly at the start time
+        //  - Or with a minimum time before the minimum time of this ignition sensor transaction
+        //
+        // TODO(efernandez) Do '&min_time = std::as_const(start_ime)' when C++17 is supported and we can use
+        //                  std::as_const: https://en.cppreference.com/w/cpp/utility/as_const
+        pending_transactions_.erase(
+            std::remove_if(pending_transactions_.begin(), pending_transactions_.end(),
+                           [&sensor_name, max_time,
+                            &min_time = start_time](const auto& transaction) {  // NOLINT(whitespace/braces)
+                             return transaction.sensor_name != sensor_name &&
+                                    (transaction.minStamp() < min_time || transaction.maxStamp() <= max_time);
+                           }),  // NOLINT(whitespace/braces)
+            pending_transactions_.end());
       }
-      // And purge out old transactions
-      //  - Either we just started and we want to purge out anything before the start time
-      //  - Or we want to limit the pending size while waiting for an ignition sensor
-      auto purge_time = ros::Time(0, 0);
-      auto last_pending_time = pending_transactions_.front().stamp();
-      if (started_)
-      {
-        purge_time = start_time;
-      }
-      else if (ros::Time(0, 0) + params_.transaction_timeout < last_pending_time)  // ros::Time doesn't allow negatives
-      {
-        purge_time = last_pending_time - params_.transaction_timeout;
-      }
-      while (!pending_transactions_.empty() && pending_transactions_.back().minStamp() < purge_time)
+      else
       {
-        pending_transactions_.pop_back();
+        // And purge out old transactions to limit the pending size while waiting for an ignition sensor
```

Comment:
Except for using `maxStamp()` instead of `minStamp()`, this is the same logic as before for the case when `started_` is still `false`.

Reply:
I'm rather indifferent on which stamp gets used here. The sole purpose of this block is to prevent the pending queue from getting too large while waiting for the ignition transaction.